airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/products_report.py

```diff
@@ -255,9 +258,17 @@ def _get_init_report_body(self, report_date: str, record_type: str, profile):
         body = {
             "reportDate": report_date,
         }
+
+# adId is automatically added to the report by amazon and requesting adId causes an amazon error
         if RecordType.ASINS in record_type:
             body["campaignType"] = "sponsoredProducts"
+            metrics_list = copy(metrics_list)
+            metrics_list.remove("adId") 
             if profile.accountInfo.type == "vendor":
                 metrics_list = copy(metrics_list)
                 metrics_list.remove("sku")
+        elif RecordType.PRODUCTADS in record_type:
+            metrics_list = copy(metrics_list)
+            metrics_list.remove("adId")
+
```

Comment:
```python
    # adId is automatically....
    if "adId" in metrics_list:
        metrics_list.remove("adId")
```

This way if you add a new report you don't need to change the code again to make it works.
Why is the reason you're using copy?

Reply:
I think that is probably better.  It does assume that all product reports have this problem but I think that is probably correct.

I made a copy because the old code that removed "sku" also made a copy and I assumed that it was probably the right thing.

Reply:
For those cases we can improve the if logic
```python
if "adId" in metrics_list and not SpecialReport:
   metrics_list.remove("adId")
```
because right now both need to remove it right?